File: rust/src/models/customer_event_data.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::customer_id::CustomerId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CustomerEventData {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub alias: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub billing_email: Option<String>,

    pub currency: String,

    /// User-defined custom property values, keyed by definition key.
    pub custom_properties: serde_json::Value,

    pub customer_id: CustomerId,

    pub invoicing_emails: Vec<String>,

    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub phone: Option<String>,
}

impl CustomerEventData {
    pub fn new(
        currency: String,
        custom_properties: serde_json::Value,
        customer_id: CustomerId,
        invoicing_emails: Vec<String>,
        name: String,
    ) -> Self {
        Self {
            alias: None,
            billing_email: None,
            currency,
            custom_properties,
            customer_id,
            invoicing_emails,
            name,
            phone: None,
        }
    }
}

File: typescript/src/models/customerEventData.ts
// this file is @generated
import { type CustomerId, CustomerIdSerializer } from "./customerId";

export interface CustomerEventData {
  alias?: string | null;

  billingEmail?: string | null;

  currency: string;

  /** User-defined custom property values, keyed by definition key. */
  customProperties: any;

  customerId: CustomerId;

  invoicingEmails: string[];

  name: string;

  phone?: string | null;
}

export const CustomerEventDataSerializer = {
  _fromJsonObject(object: any): CustomerEventData {
    return {
      alias: object["alias"],
      billingEmail: object["billing_email"],
      currency: object["currency"],
      customProperties: object["custom_properties"],
      customerId: CustomerIdSerializer._fromJsonObject(object["customer_id"]),
      invoicingEmails: object["invoicing_emails"],
      name: object["name"],
      phone: object["phone"],
    };
  },

  _toJsonObject(self: CustomerEventData): any {
    return {
      alias: self.alias,
      billing_email: self.billingEmail,
      currency: self.currency,
      custom_properties: self.customProperties,
      customer_id: CustomerIdSerializer._toJsonObject(self.customerId),
      invoicing_emails: self.invoicingEmails,
      name: self.name,
      phone: self.phone,
    };
  },
};
